### PR TITLE
Add pybind11 demo wrapper for DateTime

### PR DESCRIPTION
**Pull request by r-owen**

### CODE REVIEW

File: python/lsst/daf/base/exampleLib.cc [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        .value("TT", DateTime::Timescale::TT)
+        .export_values();
+
+    py::enum_<DateTime::DateSystem>(cls, "DateSystem")
+        .value("JD", DateTime::DateSystem::JD)
+        .value("MJD", DateTime::DateSystem::MJD)
+        .value("EPOCH", DateTime::DateSystem::EPOCH)
+        .export_values();
+
+    cls.def(py::init<long long, DateTime::Timescale>(), py::arg("nsecs") = 0LL, py::arg("scale") = DateTime::TAI)
```
Comment (r-owen) [inline python/lsst/daf/base/exampleLib.cc]: Please wrap this line

**Comment by TallJimbo** [comment]
I wonder if the switch to pybind11 might also be a good time to drop our `<moduleLib>` naming convention in favor of the broader Python convention, `_<module>`?

If you think this would require changing significantly more pure-Python code as part of the pybind11 conversion, it might be better to defer it, but I could also imagine doing it up front being easier.

File: python/lsst/daf/base/exampleLib.cc [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        .def(py::init<long long, DateTime::Timescale>())
```
Comment (r-owen) [inline python/lsst/daf/base/exampleLib.cc]: Is it practical to support keyword arguments for constructors? How about overloaded constructors?

File: python/lsst/daf/base/exampleLib.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+from _exampleLib import *
```
Comment (r-owen) [inline python/lsst/daf/base/exampleLib.py]: Please don't use from X import * in modules (other than in `__init__.py` files to hoist symbols into the package namespace). It makes it impossible to tell where symbols came from, reducing the ability of linters to work.

File: python/lsst/daf/base/exampleLib.cc (hunk 2 above)
Comment (pschella) [inline python/lsst/daf/base/exampleLib.cc]: It is practical, as long as the result doesn't become ambiguous. However, here I limit myself to only provide keyword arguments (with defaults) for the cases where the wrapped C++ call has default arguments.

File: python/lsst/daf/base/exampleLib.py (hunk 3 above)
Comment (pschella) [inline python/lsst/daf/base/exampleLib.py]: Changed. Although in the future I think we will need to make an exception for this particular case (wrapped C++ modules that need extending in Python).

Comment (r-owen) [inline python/lsst/daf/base/exampleLib.py]: I certainly hope we can avoid that. It drastically limits the ability of the python linters to detect errors, as well as making the code harder to read. If typing the full namespace becomes onerous then one can use an abbreviation.

File: python/lsst/daf/base/exampleLib.cc (hunk 2 above)
Comment (r-owen) [inline python/lsst/daf/base/exampleLib.cc]: For more readable python code and a more pythonic interface I think we will want to provide named arguments in most cases.

Comment (TallJimbo) [inline python/lsst/daf/base/exampleLib.cc]: > I think we will want to provide named arguments in most cases.

+1.  I'd think we should add kwarg support to anything with more than one argument, and maybe the ones with just one argument, too.

File: python/lsst/daf/base/exampleLib.py (hunk 3 above)
Comment (TallJimbo) [inline python/lsst/daf/base/exampleLib.py]: In the case that @pschella mentioned, I think the only modules where `from x import *` would be used would the ones where C++ classes are further specialized (like the pure-Python modules Swig generates now).  I think of these as functionally similar to `__init__` modules (neither should appear in the fully-qualified `__module__` of the classes they contain, for instance).  Those files would indeed be harder to lint, but no others would be affected.

**Comment by pschella** [comment]
Will be merged onto epic ticket.

**Pull request closed without merging**